common/core/resourcemanager.cpp

```diff
@@ -90,18 +90,22 @@ void ResourceManager::RegisterBuffer(const uint32_t& native_buffer,
 
 void ResourceManager::MarkResourceForDeletion(const ResourceHandle& handle,
                                               bool has_valid_gpu_resources) {
+  lock_.lock();
```

Comment:
hmm why is this case ?

Reply:
purged_resources_ has to be protected as it is also used in PreparePurgedResources. There could be situation where PreparePurgedResoruces are not in the same thead

Reply:
@kalyankondapally  I did not meet any bug. But I think in theory it require lock here

Reply:
ya to avoid these locks we actually separate destroy_media_resources_ and gpu resources which are accessed from multiple thread. I dont see the need for this.